app/services/search/google_search_service.rb

```diff
@@ -4,18 +4,99 @@ module Search
   class GoogleSearchService
     GOOGLE_SEARCH_URL = 'https://www.google.com/search?q=%{keyword}&hl=%{language}&lr=%{language}'
 
+    USER_AGENT_HEADER = 'User-Agent'
+    USER_AGENT = 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/113.0.0.0 Safari/537.36'
+
+    KEY_DIV_TOP_ADWORDS_CONTAINER = 'div.pla-unit-container'
+    KEY_DIV_NORMAL_ADWORDS_CONTAINER = 'div.v5yQqb'
+    KEY_DIV_NON_ADWORDS_CONTAINER = 'div.yuRUbf'
+
+    KEY_A_TAG = 'a'
+    KEY_A_TAG_ADWORDS_TOP_CLASS = 'a[data-impdclcc]'
+    KEY_A_TAG_HYPERLINK = 'a[href]'
+
+    KEY_HREF_ATTRIBUTE = 'href'
+
     def initialize(keyword, language = 'en')
       @search_url = format(GOOGLE_SEARCH_URL, keyword: keyword, language: language)
     end
 
     def search
       uri = URI.parse(@search_url)
 
-      response = Faraday.get(uri)
+      response = Faraday.get(uri, nil, USER_AGENT_HEADER => USER_AGENT)
 
       # TODO: Handle error in background job
       # https://github.com/nimblehq/ic-rails-huey-toby/issues/19
-      response.body if response.status == 200
+      @html_code = response.body if response.status == 200
+
+      parse_html(@html_code)
+    end
+
+    private
+
+    # rubocop:disable Metrics/MethodLength
+    def parse_html(html_code)
+      doc = Nokogiri::HTML(html_code)
+
+      adwords_top_urls = parse_top_adwords(doc)
+      adwords_normal_urls = parse_normal_adwords(doc)
+      non_adwords_urls = parse_non_adwords(doc)
+
+      adwords_top_count = adwords_top_urls.size
+      adwords_normal_count = adwords_normal_urls.size
+      adwords_total_count = adwords_top_count + adwords_normal_count
+      non_adwords_count = non_adwords_urls.size
+      total_links_count = parse_all_links(doc)
```

Comment:
You can create a new function to return these attributes, then we can pass it to the SearchResult initializer